daffodil-core/src/test/scala/org/apache/daffodil/general/TestPrimitives.scala

```diff
@@ -137,6 +137,131 @@ class TestPrimitives {
     TestUtils.assertEqualsXMLElements(expected, actual)
   }
 
+  @Test def testLSPDelimiter(): Unit = {
+    val sch = SchemaUtils.dfdlTestSchema(
+      <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>,
+
+      <dfdl:format ref="tns:GeneralFormat" representation="text" lengthUnits="bytes" encoding="US-ASCII" initiator="" separator="" terminator="" ignoreCase="no"/>,
+
+      <xs:element name="e1" dfdl:lengthKind="delimited">
+        <xs:complexType>
+          <xs:sequence dfdl:separator=",%LSP;" dfdl:separatorPosition="infix">
+            <xs:element name="s1" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s2" type="xs:string" dfdl:lengthKind="delimited"/>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>)
+    val (_, actual) = TestUtils.testString(sch, "SingleSpace, SingleTab,\u0009")
+    val expected: Node = <e1><s1>SingleSpace</s1><s2>SingleTab</s2></e1>
+    TestUtils.assertEqualsXMLElements(expected, actual)
+  }
+
+  @Test def testLSPPlusDelimiter(): Unit = {
+    val sch = SchemaUtils.dfdlTestSchema(
+      <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>,
+
+      <dfdl:format ref="tns:GeneralFormat" representation="text" lengthUnits="bytes" encoding="US-ASCII" initiator="" separator="" terminator="" ignoreCase="no"/>,
+
+      <xs:element name="e1" dfdl:lengthKind="delimited">
+        <xs:complexType>
+          <xs:sequence dfdl:separator=",%LSP+;" dfdl:separatorPosition="infix">
+            <xs:element name="s1" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s2" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s3" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s4" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s5" type="xs:string" dfdl:lengthKind="delimited"/>
+            <xs:element name="s6" type="xs:string" dfdl:lengthKind="delimited"/>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>)
+    val (_, actual) = TestUtils.testString(sch, "SingleSpace, MultSpace,   SingleTab,\u0009MultTab,\u0009\u0009Mix,\u0009\u0009 \u0009     \u0009End")
+    val expected: Node = <e1><s1>SingleSpace</s1><s2>MultSpace</s2><s3>SingleTab</s3><s4>MultTab</s4><s5>Mix</s5><s6>End</s6></e1>
+    TestUtils.assertEqualsXMLElements(expected, actual)
+  }
+
+  @Test def testLSPStarDelimiter(): Unit = {
+    val sch = SchemaUtils.dfdlTestSchema(
+      <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>,
+
+      <dfdl:format ref="tns:GeneralFormat" representation="text" lengthUnits="bytes" encoding="US-ASCII" initiator="" separator="" terminator="" ignoreCase="no"/>,
```

Comment:
Cannot review as this line is too long. This is true throughout this file that these dfdl:format elements are too long. 
The github review tool really gives me no way to scroll to the right here.